nixos/modules/programs/ctop.nix

```diff
@@ -0,0 +1,33 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+
+let
+  cfg = config.programs.ctop;
+in
+
+{
+  options.programs.ctop = {
+    package = mkOption {
+      type = types.package;
+      default = pkgs.ctop;
+      defaultText = "pkgs.ctop";
+      description = ''
+        The ctop package that should be used.
+      '';
+    };
+
+    enable = mkEnableOption "ctop";
+
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [
+      cfg.package
+    ];
+  };
```

Comment:
this doesn't really justify using a module. over just adding the package to `systemPackages` in the main config. a program module would be justified if other setup was necessary to use ctop, eg adding setuid wrappers or other system-wide config.

Reply:
I'll most likely add configuration options for ctop then.

Reply:
Looks like ctop doesn't support system-wide configuration. It only reads config files from `~/.config/ctop/config` and `~/.ctop`.

Reply:
maybe a home-manager module would be more useful then?

Reply:
Yeah it would. I'll leave this PR open for now.

Reply:
> I'll leave this PR open for now.

Why, if I may ask?

Reply:
Mainly to see if anyone else wants to say anything on it. I'll most likely end up closing it.

Reply:
> maybe a home-manager module would be more useful then?

The program has like 5 options you can quickly configure in the TUI. Also a home-manager module would break saving those settings which is much more inconvenient.